rosbag2/src/rosbag2/generic_subscription.cpp

```diff
@@ -98,9 +100,8 @@ GenericSubscription::borrow_serialized_message(size_t capacity)
         auto fini_return = rmw_serialized_message_fini(msg);
         delete msg;
         if (fini_return != RCL_RET_OK) {
-          RCUTILS_LOG_ERROR_NAMED(
-            "rosbag2",
-            "failed to destroy serialized message: %s", rcl_get_error_string_safe());
+          ROSBAG2_LOG_ERROR_STREAM(
```

Comment:
there is still a problem with the allocator IMO.
So assume we create a serialized message using this subscription, keep it in memory and destroy the subscription before destroying the message.
The message now has an invalid allocator.

Reply:
We had a close look and I believe it should be ok as is. The content of the allocator gets copied inside `rmw_serialized_message_init(...)` (see https://github.com/ros2/rcutils/blob/f04cc666dd06347050ff5c18b80079922e729d92/src/char_array.c#L50) so it should still be valid even after the pointer is destroyed.

We did a check by prematurely destroying the allocator (create `new rcutils_allocator_t`, use it to initialize the serialized message and then immediately destroy it) and it ran perfectly and valgrind was also happy.

Reply:
true. Please ignore my comment then.